### PR TITLE
Add build for internal flag to catch build breaks

### PR DESCRIPTION
**Pull request by j82w**
# Pull Request Template

## Description

There is a lot of conditional logic based on the internal flag. This adds a yaml file to build the solution with the internal flag set to prevent build breaks from getting merged in.

Refactored the display name to make it easier to read and prevent the values from getting truncated.

### CODE REVIEW

File: templates/build-test.yml [hunk 1]
```diff
@@ -7,6 +7,25 @@ parameters:
...
+  - checkout: self  # self represents the repo where the initial Pipelines YAML file was found
+    clean: true  # if true, execute `execute git clean -ffdx && git reset --hard HEAD` before fetching
+
+  - task: DotNetCoreCLI@2
+      displayName: Build Microsoft.Azure.Cosmos
+      inputs: 
+        command: build  
+        configuration: $(parameters.BuildConfiguration)
+        nugetConfigPath: NuGet.config
+        projects: Microsoft.Azure.Cosmos/src/Microsoft.Azure.Cosmos.csproj
```
Comment (kirankumarkolli) [inline templates/build-test.yml]: Possible to include full solution file? (i.e. UT, perf, EmulatorTests)?

Comment (j82w) [reply templates/build-test.yml]: Done

File: Microsoft.Azure.Cosmos/src/Query/Core/Metrics/IndexUtilizationInfo.cs [hunk 2]
```diff
@@ -26,7 +28,7 @@ sealed class IndexUtilizationInfo
         public IReadOnlyList<IndexUtilizationData> PotentialIndexes { get; }
 
         /// <summary>
-        /// Iniialized a new instance of the Index Utilization class.
+        /// Initialized a new instance of the Index Utilization class.
```
Comment (ealsur) [inline Microsoft.Azure.Cosmos/src/Query/Core/Metrics/IndexUtilizationInfo.cs]: Initializes?

Comment (j82w) [inline Microsoft.Azure.Cosmos/src/Query/Core/Metrics/IndexUtilizationInfo.cs]:
```suggestion
        /// Initializes a new instance of the Index Utilization class.
```

**Pull request merged**